pkg/model/openstackmodel/firewall.go

```diff
@@ -221,7 +221,7 @@ func (b *FirewallModelBuilder) addHTTPSRules(c *fi.ModelBuilderContext, sgMap ma
 	addDirectionalGroupRule(c, masterSG, nodeSG, httpsIngress)
 	addDirectionalGroupRule(c, masterSG, masterSG, httpsIngress)
 
-	if b.UseLoadBalancerForAPI() {
```

Comment:
Is it possible to continue to honor `UseLoadBalancerForAPI`?  This might be more consistent across clouds...

Reply:
I believe I might be able to do this by addressing your previous comment

Reply:
It looks like my issues revolve around using public topology without a loadbalancer, as this will initialize the loadbalancer struct.  

My understanding of public topology for openstack might mean that either a loadbalancer is created and exposed outside of the openstack environment or floating IP's of the masters, without loadbalancers, are exposed.

My understanding of private topology might mean for me that either a loadbalancer with only local access is created for use by kubelet, or in the event that an openstack does not support loadbalancers gossip or designate can be used.

@zetaab do you have any opinions on this.  Unless I am misunderstanding how this ought to work I cannot use this method as I am under the impression that we could do a public and private topology without the use of loadbalancers, however undesirable it may be.

Reply:
This default openstack setup is kind of "private" if we think how kops works in aws, or aws private subnets works itself. In my opinion in case of openstack the public and private from network and subnet are similar, the only difference is that is api exposed to outside or not? Should we even have terminology private and public in openstack kops?

Reply:
@justinsb any thoughts here?

I could potentially allow private topology to create a loadbalancer and expose it outside, which seems weird to me.  However I cannot specify public topology and use this method to determine if a loadbalancer should be created or not, as this method will always return true under public topology.

Reply:
I am under the opinion that we remove support for Kops without loadbalancers.  This isn't an elegant way to deploy kubernetes.